bind.go

```diff
@@ -60,9 +60,14 @@ func (l *Conn) SimpleBind(simpleBindRequest *SimpleBindRequest) (*SimpleBindResu
 	}
 	defer l.finishMessage(messageID)
 
-	packet = <-channel
-	if packet == nil {
-		return nil, NewError(ErrorNetwork, errors.New("ldap: could not retrieve response"))
+	packetResponse, ok := <-channel
+	packet, err = packetResponse.ReadPacket()
```

Comment:
alternately, don't check `ok` at all and trust the `nil` receiver impl of `PacketResponse`